examples/components/MNIST/traininsilo/run.py

```diff
@@ -77,6 +88,28 @@ def __init__(
             self.test_dataset_, batch_size=batch_size, shuffle=True
         )
 
+        # DP
+        logger.info(f"DP: {dp}")
+        if dp:
+            if not ModuleValidator.is_valid(self.model_):
+                self.model_ = ModuleValidator.fix(self.model_)
+
+        self.optimizer_ = SGD(self.model_.parameters(), lr=lr, momentum=0.9)
+
+        if dp:
+            privacy_engine = PrivacyEngine(secure_mode=False)
```

Comment:
nit: might be good to add some doc strings to explain "secure_mode" and point out potential pitfall about it. Note that, using secure_mode=False is better for performance (i.e., faster) but it has known weakness/vulnerability that doesn't meet the safety bar for scenarios targeting for shipping model to production. So it is good for faster exploration/experimenting, but need to turn on (i.e., secure_model=True) when training DP model to ship to production. Btw, below I shared official docstring explanation about secure_mode:

secure_mode: Set to ``True`` if cryptographically strong DP guarantee is
                required. ``secure_mode=True`` uses secure random number generator for
                noise and shuffling (as opposed to pseudo-rng in vanilla PyTorch) and
                prevents certain floating-point arithmetic-based attacks.
                See :meth:`~opacus.optimizers.optimizer._generate_noise` for details.
                When set to ``True`` requires ``torchcsprng`` to be installed

Reply:
makes sense! done